src/main/java/com/internet/shop/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.internet.shop.security.impl;
+
+import com.internet.shop.exception.AuthenticationException;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.User;
+import com.internet.shop.security.AuthenticationService;
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect username or password"));
+        if (user.getPassword().equals(password)) {
```

Comment:
let's check all conditions inside `if` so that we could throw our Exception only once

Reply:
Done